merlin/models/tf/blocks/core/transformations.py

```diff
@@ -556,3 +556,76 @@ def call_outputs(
         targets = transform_label_to_onehot(targets, num_classes)
 
         return outputs.copy_with_updates(targets=targets)
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin_models")
+class PopularityLogitsCorrection(Block):
+    """Correct the predicted logit scores based on the item frequency
+
+    This first implementation requires the user to provide
+    the item frequency table.
+
+    TODO: Use the schema to infer the path to the item frequency parquet table
+
+    Parameters:
+    ----------
+    item_frequency : Union[tf.Tensor]
+        The item frequency table
+    """
+
+    def __init__(self, item_frequency: Union[tf.Tensor, str], **kwargs):
+        super().__init__(**kwargs)
+        candidate_probabilities = tf.cast(
+            item_frequency / tf.reduce_sum(item_frequency), tf.float32
+        )
+        self.candidate_probabilities = tf.Variable(
```

Comment:
I think we should also have an `update()` method, in case we need to update the item popularities when fine tuning a pre-trained model